### PR TITLE
Add new address generation tests

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Added new address generation tests with mnemonic and Stronghold secret managers and also with message_interface

## Links to any relevant issues

#423 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

With the test

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/fixtures/test_vectors.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
+{
```
Comment (thibault-martinez) [inline tests/fixtures/test_vectors.json:1]: What was used to generate these fixtures ?

Comment (Thoralf-M) [reply tests/fixtures/test_vectors.json:1]: https://thoralf-m.github.io/iota-ed25519-address-wasm/ :trollface:

Comment (thibault-martinez) [reply tests/fixtures/test_vectors.json:1]: And what does this use under the hood ? Ideally fixtures need to be generated from a completely different implementation.

Comment (Thoralf-M) [reply tests/fixtures/test_vectors.json:1]: crypto.rs and bee-message
It's also compared with hornet from https://github.com/gohornet/hornet/blob/develop/private_tangle/private_tangle_keys.md and https://github.com/iotaledger/tips/blob/add-smr-hrps/tips/TIP-0031/tip-0031.md and iota.c https://github.com/iotaledger/iota.c/blob/1a4563d0d0750fe5b16fbaae71142c9f232e303c/tests/wallet/test_wallet.c#L83-L113

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**